cpp/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -629,6 +629,7 @@ inline Status ConvertStruct(const PandasOptions& options, const ChunkedArray& da
   // preserve the <= 0.15.1 behavior until a better solution can be devised
   PandasOptions modified_options = options;
   modified_options.ignore_timezone = true;
+  modified_options.coerce_temporal_nanoseconds = false;
```

Comment:
This was needed because for the `ConvertArrayToPandas` path I now set `coerce_temporal_nanoseconds` to `True`. 

This actually highlighted an inconsistency in this struct conversion between array vs table: for Table.to_pandas, this was still giving integers (because `coerce_temporal_nanoseconds` is set to True in that path). Expanded the tests to cover this as well.